portfolio/src/main/webapp/script.js

```diff
@@ -42,10 +42,52 @@ function fadeOut(){
 }
 
 //From Week-3 Tutorial
-function getGreeting() {
-  console.log('Handling the response.');
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementsByClassName('greeting-cont')[0].innerText = quote;
+function getComments() {
+  //Get the number of comments to show per page
+  numComments = localStorage.getItem("num-of-comments");
+  fetch(`/comments?num-comments=${numComments}`).then(response => response.json()).then((comment) => {
+    container = document.getElementsByClassName('comments-section')[0];
+    container.innerHTML = "";
+    for (item in comment) {
+        container.appendChild(createListElement(comment[item]));
+    }
   });
+  document.getElementsByName('num-comments')[0].value = numComments;
 }
 
+/** Creates an <li> element containing text. */
+function createListElement(comment) {
+  const liElement = document.createElement('li');
+  liElement.className += "comment-list";
+  
+  fetch("/comment.html").then(response=>response.text()).then(data =>{
+      liElement.innerHTML = data;
+      liElement.getElementsByClassName("comment-words")[0].innerText = comment.content;
+      //Add funcitonality to the delete button
+      deleteButtonElement = liElement.getElementsByClassName('comment-delete')[0];
+      deleteButtonElement.addEventListener('click', () => {
+      deleteComment(comment);
+
+      // Remove the comment from the DOM.
+      liElement.remove();
+    });
+    liElement.getElementsByClassName("comment-date")[0].innerText = comment.date;
+      
+  });
+
+  return liElement;
+}
+
+function amountOfComments(){
+    //The program will remember the number of comments the user wants to display
+    num = document.getElementsByName('num-comments')[0].value;
+    localStorage.setItem("num-of-comments", num);
+    getComments();
+}
+
+function deleteComment(comment) {
+  //Remove a comment and then reload the number of comments
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-data', {method: 'POST', body: params}).then(getComments());
```

Comment:
I think this should be

```
.then(() => getComments());
```
since we want to actually defer the evaluation of `getComments()`, since `.then()` needs to take a callback function

Reply:
done